src/pos/tasks/ot_croissant.py

```diff
@@ -0,0 +1,93 @@
+
+import json
+from datetime import datetime
+
+from loguru import logger
+
+from ot_croissant.crumbs.metadata import PlatformOutputMetadata
+
+
+class OtCroissantError(OtterError):
+    """Base class for exceptions in this module."""
+
+class OtCroissantSpec(Spec):
+    """Configuration fields for the OT Croissant task.
+
+    This task has the following custom configuration fields:
+        - ftp_address (str): The URL of the ftp where the data is going to be published.
+        - gcp_address (str): The URL of the google bucket where the data is going to be published.
+        - dataset_path (Path): The path where the parquet outputs are stored. These outputs are going to be used to extract the schema.
+        - date_published (str): The date when the data was published. The date format is YYYY-MM-DD.
+        - output (str): Path (relative to `work_path` or `release_uri`) to store the metadata at.
+    """
+
+    ftp_address: str
+    gcp_address: str
+    dataset_paths: list[Path]
+    date_published: str
+    output: str
+
+def datetime_serializer(obj):
+    if isinstance(obj, datetime):
+        return obj.isoformat()
+    raise TypeError(f"Type {type(obj)} not serializable")
+
+class OtCroissant(Task):
+    def __init__(self, spec: OtCroissantSpec, context: TaskContext) -> None:
+        super().__init__(spec, context)
+        self.spec: OtCroissantSpec
+        self.local_path: Path = self.context.config.work_path / self.spec.output
+        self.remote_uri: str | None = None
+        if context.config.release_uri:
+            self.remote_uri = f'{context.config.release_uri}/{spec.output}'
+
+    @report
+    def run(self) -> Self:
+
+        release = self.context.scratchpad.sentinel_dict.get('release')
+        if not release:
+            raise ScratchpadError('"release" not found in the scratchpad')
+
+        # Converting the list of paths to a list of strings and prepending the work_path
+        logger.debug("Converting the list of paths to a list of strings and prepending the work_path")
+        datasets = list(map(lambda path : str(self.context.config.work_path.joinpath(path)), self.spec.dataset_paths))
+
+        logger.debug(f"Generating metadata for release {release}")
+        metadata = PlatformOutputMetadata(
+            datasets=datasets,
+            ftp_location= self.spec.ftp_address,
+            gcp_location=self.spec.gcp_address,
+            version=release,
+            date_published=self.spec.date_published,
+            data_integrity_hash='sha256'
+        )
+        logger.debug(f"Metadata generated: {metadata}")
+
+        with open(self.local_path, "w+") as f:
+            logger.debug(f"Writting metadata to {self.local_path}")
+            metadata_json = metadata.to_json()
+            metadata_str = json.dumps(metadata_json, indent=2, default=datetime_serializer)
+            content = f'{metadata_str}\n'
+            f.write(content)
+            logger.debug("Metadata written")
```

Comment:
```suggestion
        with open(self.local_path, "w+") as f:
            json.dump(dataset_paths, f, indent=2)
            f.write("\n")
        logger.debug("metadata written successfully to {self.local_path}")
```

you can do this with `json.dump()` straight away without using a string first.